examples/gtest_emulation/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2020  ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.13)
+
+project(gtest CXX)
+
+if(EXISTS "${HPX_DIR}")
+
+  find_package(HPX REQUIRED)
+
+  # Add a static library which contains a main to emulate gtest_main
+  add_library(static_main_lib STATIC static_main.cpp)
+
+  # /!\ This helper interface is needed to keep the right linking order
+  add_library(hpx_helper_interface INTERFACE)
+  target_link_libraries(
+    hpx_helper_interface INTERFACE HPX::hpx HPX::wrap_main static_main_lib
+  )
+
+  # Test with the main function in a separate static library
+  add_executable(hpx_main_ext_main hpx_main_ext_main.cpp)
+  target_link_libraries(hpx_main_ext_main PRIVATE hpx_helper_interface)
+
+else()
+  message(
+    WARNING
+      "HPX_DIR=${HPX_DIR} does not exist. Did you forget to run the install rule?"
+  )
```

Comment:
Shouldn't things work from the build directory as well?

Reply:
This is unrelated to that... This just prints a warning if `HPX_DIR` (i.e. the install directory) is not found. The build directory is always found. We discussed this a long time ago and left it as a warning since that means one can run the build tests also without installing HPX.

Reply:
So there is actually two targets constructed as for the hello world component, one for the build directory and one for the install directory. I would like for the test to be the closest to the real case (i.e. most probably a project using an installed  version of HPX), what do you think?

Reply:
You can check what windows does with `cmake --target install`. It would indeed be good to test the installed version as well.

Reply:
> This is unrelated to that... This just prints a warning if `HPX_DIR` (i.e. the install directory) is not found. The build directory is always found. We discussed this a long time ago and left it as a warning since that means one can run the build tests also without installing HPX.

Why do we need a warning if everything works?

Reply:
There are four options:
1. Silently ignore it if `HPX_DIR` does not exist (i.e. HPX has not been installed).
2. Print a warning.
3. Make it an error.
4. Make the test depend on the `install` target.
We currently do number 2 because it means that tests don't fail if you haven't installed HPX, but you still get a reminder that this test essentially was not run. We can change this if you think one of the other options would be better.

Reply:
I'm fine with the warning, however Id suggest to make it more explanatory - e.g. saying that the test such-and-such was not run because... Or similar. 

BTW, can we use `hpx_warning()` instead of `message()`?

Reply:
> BTW, can we use `hpx_warning()` instead of `message()`?

Unfortunately not, because in this case HPX is not found. This CMakeLists.txt is for an "external project".